app/actions/service_instance_update_managed.rb

```diff
@@ -129,5 +132,24 @@ def raise_if_bind_inconsistency!(service_instance, message)
         'cannot switch to non-bindable plan when service bindings exist'
       )
     end
+
+    def raise_if_invalid_maintenance_info_change!(service_instance, message)
+      return unless message.maintenance_info_version
+
+      raise_if_unsupported_by_current_plan!(service_instance, message)
+      raise_if_version_mismatch!(service_instance, message)
+    end
+
+    def raise_if_unsupported_by_current_plan!(service_instance, message)
+      return if service_instance.service_plan.maintenance_info
```

Comment:
This checks MI against the current plan, whereas it's more correct to check against the plan you're moving to. But this story is just about updating the MI by itself, so this is ok for now.

Reply:
hmmm.. anyway, apparently you cannot change plan and maintenance_info at the same time as per [#172861407](https://www.pivotaltracker.com/story/show/172861407). Will check what v2 does anyway.